### DISCUSSION TITLE
What is the browser compatibility?

### DISCUSSION THREAD

**Discussion by zhanglolo** [unocss/unocss#2234] [Q&A]
What is the browser compatibility?

**Comment by StarLederer** [comment]
UnoCSS engine is used on the server side or at build time in most cases so it is likely that you don't need to be concerned about the engine's browser comparability. If you mean the default preset (@unocss/preset-uno) or any of the presets it inherits from, I am not exactly sure but it should be very close to what is described on the [Tailwind's page about this](https://tailwindcss.com/docs/browser-support).

Even if there are utilities for things that your target platform does not support you can simply not use them. To confirm comparability you can see what CSS is generated using the [interactive docs](https://uno.antfu.me/) and check the compatibility on [caniuse](https://caniuse.com/).

You can also look for community presets that target older platforms or even make your own. If you really want to make sure your team does not use features that are too modern you can also make your own preset that produces errors when those rules are used or, the opposite, make your own preset (optionally based on one of the [official ones](https://github.com/unocss/unocss#official-presets)) that only reexports the things that you want to allow.

**Comment by leosin** [comment]
CSS Layers is a new CSS feature for managing style priorities. Here is its browser compatibility information:

CSS Layers Compatibility
Chrome: Version 99 and above
Firefox: Version 97 and above
Safari: Version 15.4 and above
Edge: Version 99 and above

Usage Recommendations

When using CSS Layers, make sure the target browser version supports this feature.
You can define style layers through the @layer directive to increase the flexibility of style management.
For unsupported browsers, consider providing fallback styles or using polyfills.
Make sure the target browser version of your project meets these requirements to take full advantage of CSS Layers.

**Comment by ntnyq** [comment]
Today I found UnoCSS generated gradient color failed to render on Chrome v101. 
https://unocss.dev/presets/legacy-compat#legacy-compat-preset is required to provider a better compatibility. You need to install it manually.
